react-and-redux-style-guide/README.md

```diff
@@ -0,0 +1,358 @@
+> Reason: this makes the connection between a React component and Redux storage more transparent and isolated.
+```typescript
+// EXAMPLE
+// /someModule/containers/FilterContainer.tsx
+import {Filter} from '../components/settings/Filter';
+import {itemsSelector} from '../redux/selectors';
+import {updateFilter} from '../redux/actions';
+
+const mapStateToProps = (state) => ({
+    items: itemsSelector(state),
+});
+
+const mapDispatchToProps = {
+    onChange: updateFilter,
+};
+
+export const FilterContainer: React.FC<{}> = connect(
+    mapStateToProps,
+    mapDispatchToProps,
+)(Filter);
+
+```
+## 7. Add typings for actions
+```typescript
+// EXAMPLE
+const UPDATE_USER = 'users/updateUser';
+const REMOVE_USER = 'users/removeUser';
+
+export interface Action<Type = string, Payload = any> {
+    type: Type;
+    payload?: Payload;
+}
+
+export type UpdateUserAction = Action<
+    typeof UPDATE_USER, 
+    {
+        id: number;
+        name: string;
+    }
+>;  
+
+export type RemoveUserAction = Action<typeof REMOVE_USER, number>;
+
+export const someReducer: Reducer<UsersStore, UpdateAction | RemoveAction> = (
+    state = initialState,
+    action,
+) => { ... };
+```
+## 8. Minimize the use of "blind spreads/returns" 
+> Reason: this improves readability of a reducer
+```typescript
+// BAD
+...
+    case ACTION: {
+        return {...state, ...action.payload}
+    }
+... 
+// BAD
+...
+    case ACTION: {
+        return action.payload
+    }
+...  
+```
+```typescript
+// GOOD
+...
+    case ACTION: {
+        let {id, item} = action.payload;
+        return {
+            ...state,
+            id,
+            item,
+        }
+    }
+... 
+``` 
+## 9. Always write unit tests for reducers
+> Reason: a reducer is one of the most sensitive places of the redux code, a modification can have a lot of impact on the application. On the other hand, this code is relatively easy to test, so writing unit tests is a convenient way to be sure the reducer works as expected, and some changes don't affect it. 
+```typescript
+// EXAMPLE
+import {rolesReducer} from '../rolesReducer';
+const ROLES: SomeRole[] = [{ ... }, ...]; 
+...
+    it('should handle FETCH_ACTIVE_ROLES by replacing roles', () => {
+        const actualState = rolesReducer(INITIAL_STATE, {
+            type: FETCH_ACTIVE_ROLES,
+            payload: ROLES,
+        });
+        expect(actualState.roles).toEqual(ROLES);
+    });
+```
+## 10. Keep only action names in `/moduleName/redux/constants.ts` file.
+```typescript
+// BAD
+// src/app/modules/extensions/redux/constants.ts
+export const FETCH_EXTENSIONS = 'extensions/fetch';
+export const EXTENSIONS_MAIN_CONSTANT_VALUE = 42;
+...
+```
+```typescript
+// GOOD
+// src/app/modules/extensions/redux/constants.ts
+export const FETCH_EXTENSIONS = 'extensions/fetch';
+
+// src/app/modules/extensions/constants.ts
+export const EXTENSIONS_MAIN_CONSTANT_VALUE = 42;
+```
+## 11. Write Action Types as "domain/eventName"
```

Comment:
Let's lower case it "Write Action Types ..." > "Write action types ..." ?

Reply:
done